lib/deflate.rb

```diff
@@ -0,0 +1,23 @@
+class Array
+  def deflate(*args, &block)
+    original = dup
```

Comment:
This is interesting. Could you take me through why you chose to `dup` the array?